### DISCUSSION TITLE
Hotkey to Disconnect/Connect Controller...

### DISCUSSION THREAD

**Discussion by IstanbulRed** [Valkirie/HandheldCompanion#335] [Ideas]
First of all...THANK YOU SO MUCH for this piece of software. It is an absolute godsend for my SteamDeck. It has dramatically improved my user experience. It truly is sensational. I love it so much. The Dualshock implementation is my favourite as I am a big PS Remote Play user - and of course the hotkeys are just a must have feature!

But my one idea;

- Could we assign a hotkey to connect/disconnect the controller?
- This would be useful to me as when I play my SteamDeck docked, I use an external controller and sometimes it can confuse whatever game/app I am using if the SteamDeck controls are still active . I can open the Handheld Companion and quickly press disconnect/connect of course, but assigning this action to a hotkey would further improve my quality of life when using the Deck.

But again, thank you for this excellent application!!!

**Comment by CasperH2O** [comment]
Thank you for the kind words @IstanbulRed .

If you have this as a hotkey, how will you use the hotkey when your controller is not yet connected and when you do have it connected won't it disconnect then, forcing you to get your SD regardless? 🤔

**Reply by IstanbulRed** [reply]
Apologies if I haven't thought about this correctly. This is the way I use now:

1. Handheld Companion Service runs all the time.
2. I have a DS4 'connected' in Handheld Companion. This covers all my needs when playing the Steamdeck in handheld mode.
3. I have Lizard buttons and mouse mapped to two back Steamdeck buttons if I need them.

4. When I want to use the SteamDeck docked and connected to my TV, and I want to use external controllers... I go into the HC and 'disconnect' the controller. This then removes the controller from Windows.
5. I then plug in my external controllers or connect them via Bluetooth. 
6. Some games and app automatically set the DS4 to player 1 when it is still connected in the HC app, so you then can't use the external controllers properly. 
7. When I'm finished playing on my TV, I unplug all external controllers, then pick up my Steamdeck, open the HC app and then 'connect' the virtual Dualshock 4 again - ready for handheld play. 

Hopefully I'm making sense.

**Reply by CasperH2O** [reply]
Yes. And in between which steps would you like to use the hotkey instead and on what device (external controller or SteamDeck)?

**Reply by IstanbulRed** [reply]
Hotkey would be on the Steamdeck.

Step 4 to disconnect.

Step 7 to reconnect.

I'm literally talking about the option you have below the picture of the controller on the controller tab. Where you can select 'connected' or 'disconnected'. If that could be a toggle on a hotkey.
